File: ruby/pgms_ruby/09-pgmAvionInterface/IAvionStandar.rb
require_relative 'IAvionCarga'
require_relative 'IAvionMilitar'
require_relative 'IAvionGranja'

class IAvionStandar
  include IAvionCarga
  include IAvionMilitar
  include IAvionGranja
  
  def startingEngine
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  def stopingEngine
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  def speedUp
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  def currentSpeed
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  def landing
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  def descent
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  def chargeFuel(litros)
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  def setMarca(marca)
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  def setTipo(tipo)
    raise NotImplementedError, "#{self.class} no ha implementado el método '#{__method__}'"
  end
  
  private :startingEngine, :stopingEngine, :speedUp, :currentSpeed, :landing, :descent, :chargeFuel, :setMarca, :setTipo
end

File: python/java_to_python/09_pgm_avion_interface/IAvionStandar.py
from abc import ABC, abstractmethod
from IAvionCarga import IAvionCarga 
from IAvionMilitar import IAvionMilitar
from IAvionGranja import IAvionGranja

class IAvionStandar(IAvionCarga, IAvionMilitar, IAvionGranja, ABC):
    
    @abstractmethod
    def startingEngine(self):
        pass
    
    @abstractmethod
    def stopingEngine(self):
        pass
    
    @abstractmethod
    def speedUp(self):
        pass
    
    @abstractmethod
    def currentSpeed(self):
        pass
    
    @abstractmethod
    def landing(self):
        pass
    
    @abstractmethod
    def descent(self):
        pass
    
    @abstractmethod
    def chargeFuel(self, litros):
        pass
    
    @abstractmethod
    def setMarca(self, marca):
        pass
    
    @abstractmethod
    def setTipo(self, tipo):
        pass